src/messageHandler.ts

```diff
@@ -48,36 +66,61 @@ export class MessageHandler {
   /**
   * You can use this method if you need to manually update the token.
   */
-  postUpdateToken = (requestId: string, token: string) => {
-    const message = this.buildMessage(messages.updateToken, requestId, token);
+  postUpdateToken = (message: UpdateTokenMessage) => {
+    const event: UpdateTokenEvent = {
+      name: partnerEvents.updateToken,
+      message
+    }
 
-    this.postMessage(message);
+    this.postMessage(event);
   }
 
   /**
   * You can use this method if you need to manually inform the dreams app that account provision has been initiated.
   */
-  postAccountProvisionInitiated = (requestId: string) => {
-    const message = this.buildMessage(messages.accountProvisioned, requestId);
+  postAccountProvisionInitiated = (message: Message) => {
+    const event: AccountProvisionInitiatedEvent = {
+      name: partnerEvents.accountProvisionInitiated,
+      message
+    }
+
+    this.postMessage(event);
+  }
+
+  /**
+  * You can use this method if you need to manually inform the dreams app that investment account provision has been initiated.
+  * DreamId should match the one received initially with the onInvestmentAccountProvisionRequested event.
```

Comment:
needed?

Reply:
check my comment above -> IMO we need to expose it. Happy to have a chat about it though